File: clients/Java-GraalVM-native/src/main/java/model/Action.java
package model;

import util.StreamUtil;

public class Action {
    private java.util.Map<Integer, model.EntityAction> entityActions;
    public java.util.Map<Integer, model.EntityAction> getEntityActions() { return entityActions; }
    public void setEntityActions(java.util.Map<Integer, model.EntityAction> entityActions) { this.entityActions = entityActions; }
    public Action() {}
    public Action(java.util.Map<Integer, model.EntityAction> entityActions) {
        this.entityActions = entityActions;
    }
    public static Action readFrom(java.io.InputStream stream) throws java.io.IOException {
        Action result = new Action();
        int entityActionsSize = StreamUtil.readInt(stream);
        result.entityActions = new java.util.HashMap<>(entityActionsSize);
        for (int i = 0; i < entityActionsSize; i++) {
            int entityActionsKey;
            entityActionsKey = StreamUtil.readInt(stream);
            model.EntityAction entityActionsValue;
            entityActionsValue = model.EntityAction.readFrom(stream);
            result.entityActions.put(entityActionsKey, entityActionsValue);
        }
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, entityActions.size());
        for (java.util.Map.Entry<Integer, model.EntityAction> entityActionsEntry : entityActions.entrySet()) {
            int entityActionsKey = entityActionsEntry.getKey();
            model.EntityAction entityActionsValue = entityActionsEntry.getValue();
            StreamUtil.writeInt(stream, entityActionsKey);
            entityActionsValue.writeTo(stream);
        }
    }
}

File: clients/Kotlin-GraalVM-native/src/main/kotlin/model/Action.kt
package model

import util.StreamUtil

class Action {
    lateinit var entityActions: MutableMap<Int, model.EntityAction>
    constructor() {}
    constructor(entityActions: MutableMap<Int, model.EntityAction>) {
        this.entityActions = entityActions
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Action {
            val result = Action()
            val entityActionsSize = StreamUtil.readInt(stream)
            result.entityActions = mutableMapOf()
            for (i in 0 until entityActionsSize) {
                var entityActionsKey: Int
                entityActionsKey = StreamUtil.readInt(stream)
                var entityActionsValue: model.EntityAction
                entityActionsValue = model.EntityAction.readFrom(stream)
                result.entityActions.put(entityActionsKey, entityActionsValue)
            }
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, entityActions.size)
        for (entityActionsEntry in entityActions) {
            StreamUtil.writeInt(stream, entityActionsEntry.key)
            entityActionsEntry.value.writeTo(stream)
        }
    }
}
